tools/release.py

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env python3
+
+import argparse
+import logging
+import re
+import sys
+
+from utils import (
+    run,
+    repository_dirty,
+    remote_exists,
+    matching_remote_branches,
+    local_tag_exists,
+    remote_tag_exists,
+    local_branch_matches_remote_branch,
+    configure_git_user,
+    create_local_tag,
+    push_tag,
+)
+
+log = logging.getLogger(__name__)
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s %(levelname)s %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%SZ",
+)
+
+SEMVER_MAJOR_MINOR_PATTERN = "(0|[1-9]\d*)\.(0|[1-9]\d*)"
+
+# https://regex101.com/r/vkijKf/1/
+SEMVER_PATTERN = "(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?"
+
+RELEASE_TAG_PATTERN = f"v{SEMVER_PATTERN}-{SEMVER_PATTERN}"
```

Comment:
Use `re.compile` here and in the next line so it's validated early and ready to use below.

Reply:
Are you happy with this? https://github.com/mesosphere/kudo-cassandra-operator/pull/26/commits/99a0ac016b90b3d30da539ce86e21e6fde51012a

To be honest, I slightly prefer the previous version. But, it can go either way.